eng/templates/steps/get-build-info.yml

```diff
@@ -66,63 +82,29 @@ steps:
-          exit 1
-        fi
-
-        installer_official_run_id=$(echo $installer_official_run_ids | jq -r '.[0]')
+        source eng/get-build-info.sh
 
-        installer_tarball_build_runs=$(az pipelines runs list --organization $(azDoOrg) --project $(azDoProject) --pipeline-ids $(installerTarabllBuildCiPipelineId))
-        installer_tarball_build_run_ids=$(echo $installer_tarball_build_runs | jq -r '[.[] | select(.sourceVersion == "$(InstallerCommit)") | { "result": .result, "id": .id }]')
-        num_tarball_build_runs=$(echo $installer_tarball_build_run_ids | jq 'length')
-
-        if [ "$num_tarball_build_runs" != "1" ]; then
-          echo "##vso[task.logissue type=error]There are ${num_tarball_build_runs} runs of dotnet-installer-source-build-tarball-build for $(InstallerCommit). Please manually specify run IDs to use."
-          echo "##vso[task.logissue type=error]Run IDs are: ${installer_tarball_build_run_ids}"
-          exit 1
+        if [[ "$(ReleaseChannel)" == "8.0" ]]; then
+            get_build_info "$(azDoOrg)" "$(azDoProject)" "$(dotnetDotnetCiPipelineId)" dotnet-dotnet DotnetDotnetRunId "installer-$(InstallerCommit)"
+        else
+            get_build_info "$(azDoOrg)" "$(azDoProject)" "$(installerOfficialCiPipelineId)" dotnet-installer-official-ci InstallerOfficialRunId
+            get_build_info "$(azDoOrg)" "$(azDoProject)" "$(installerTarballBuildCiPipelineId)" dotnet-installer-source-build-tarball-build InstallerTarballBuildRunId
         fi
-
-        installer_tarball_build_run_id=$(echo $installer_tarball_build_run_ids | jq -r '.[0].id')
-        installer_tarball_build_run_result=$(echo $installer_tarball_build_run_ids | jq -r '.[0].result')
-
-        if [[ ("$installer_tarball_build_run_result" == "failed") && "$(useSpecificPipelineRunIDs)" ]]; then
-          echo "##vso[task.logissue type=error]: dotnet-installer-source-build-tarball-build run ID ${installer_tarball_build_run_id} failed. Please manually specify a tarball build ID to use instead. Exiting..."
-          exit 1
-        fi
-
-        echo "dotnet-installer-official-ci run ID: ${installer_official_run_id}"
-        echo "  Link: https://dev.azure.com/dnceng/internal/_build/results?buildId=${installer_official_run_id}"
-        echo "dotnet-installer-source-build-tarball-build run ID: ${installer_tarball_build_run_id}"
-        echo "  Link: https://dev.azure.com/dnceng/internal/_build/results?buildId=${installer_tarball_build_run_id}"
-
-        echo "##vso[task.setvariable variable=InstallerOfficialRunId]${installer_official_run_id}"
-        echo "##vso[task.setvariable variable=InstallerTarballBuildRunId]${installer_tarball_build_run_id}"
       name: GetAssociatedPipelineRunsStep
       displayName: Get Associated Pipeline Run IDs
       env:
         AZURE_DEVOPS_EXT_PAT: $(System.AccessToken)
 
-  - ${{ if eq(parameters.useSpecificPipelineRunIDs, true) }}:
+  - ${{ else }}:
```

Comment:
Given the introduction of the get_build_info script, the if and else bodies have more in common that differ.  Have you considered simply conditionalizing on parameters.useSpecificPipelineRunIDs within the script body?  If feels like it would simplify the logic more.

Reply:
I tried but there's too many differences so I think it's actually more readable like this